doc/changes/devel/12430.bugfix.rst

```diff
@@ -0,0 +1 @@
+Reformats channel and detector lookup in :class:`mne.io.RawSNIRF` from array based to dictionary based. Removes incorrect assertions that every detector and source must have data associated with every registered optode position, by :newcontrib:`Alex Kiefer`.
```

Comment:
An odd thing -- we don't actually document that class so Sphinx will get mad about that ref. We document the reader, then point people to `mne.io.Raw`. So best just to ref the reading function
```suggestion
Reformats channel and detector lookup in :func:`mne.io.read_raw_snirf` from array based to dictionary based. Removes incorrect assertions that every detector and source must have data associated with every registered optode position, by :newcontrib:`Alex Kiefer`.
```